### PR TITLE
Rescue and log email sending errors

### PR DESCRIPTION
**Pull request by cantino**
Replaces #1300 with a more thread-safe appracoh.

@dsander, see any downside to just sending emails immediately instead of using jobs since we're inside a job anyway?

### CODE REVIEW

**Comment by jevanpipitone** [comment]
Hi I am not advanced enough with Ruby on Rails to be able to check your code so hopefully someone else can check it for you. If you use deliver_now will it freeze the UI until it is delivered?

File: app/models/agents/email_agent.rb [hunk 1]
```diff
@@ -37,19 +37,27 @@ def default_options
...
+              from: interpolated(event)['from'],
+              subject: interpolated(event)['subject'],
+              headline: interpolated(event)['headline'],
+              body: interpolated(event)['body'],
+              content_type: interpolated(event)['content_type'],
+              groups: [present(event.payload)]
+            ).deliver_now
+            log "Sent mail to #{recipient} with event #{event.id}"
+          rescue => e
+            error("Error sending mail to #{recipient} with event #{event.id}: #{e.message}")
```
Comment (dsander) [inline app/models/agents/email_agent.rb]: Should we re-raise the exception so the job can be retried?

Comment (dsander) [inline app/models/agents/email_agent.rb]: If we re-raise it might be good to add `no_bulk_receive!`

**Comment by dsander** [comment]
@cantino I think that should not be an issue, we only lose the ability to retry each delivery job.

**Comment by cantino** [comment]
How important do you think that ability is?

File: app/models/agents/email_agent.rb (hunk 1 above)
Comment (cantino) [inline app/models/agents/email_agent.rb]: That's a good idea; I guess if we raise, we won't update the memory, and it should retry cleanly.

**Comment by dsander** [comment]
It's probably not a huge deal, if the SMTP server is available for one recipient it should also work for the rest of them. Having errors in the agent log outweighs that downside in my opinion.

**Comment by cantino** [comment]
I added re-`raise` commands and manually checked to see that they get retried and logged.

Looks like the EmailAgent already had `no_bulk_receive!`.  I don't think the EmailDigestAgent needs it because emails are sent on `check`, not `receive`.

Mind giving this a final look-over?

**Comment by cantino** [comment]
We're starting to see flaky phantom specs, which is annoying.  This is otherwise green.

**Comment by dsander** [comment]
Nice! Looks good to me. I will have a look at the capybara/poltergeist configuration, increasing the timeout should be simple.

**Pull request merged**